draft-private-access-tokens.md

```diff
@@ -416,15 +440,91 @@ Issuance assumes the Client has the following information, derived from a given
-and Origin. In particular, for each pair, Issuers maintain a stable mapping from ANON_CLIENT_ID
-to ORIGIN_NAME and ANON_ORIGIN_ID_PRIME values, as well as policy state about tokens
-that have already been issued to a Client. Policy state is custom to each implementation:
-it could include a simple counter to track the number of tokens issued to a
-ANON_CLIENT_ID, or have more detailed timestamp information.
+One possible mechanism for implementing this identifier is for the Client to store a mapping
+between the ORIGIN_NAME and a randomly generated ANON_ORIGIN_ID for future requests. Alternatively,
+the Client can compute a PRF keyed by a per-client secret (CLIENT_SECRET) over the ORIGIN_NAME,
+e.g., ANON_ORIGIN_ID = HKDF(secret=CLIENT_SECRET, salt="", info=ORIGIN_NAME).
 
-Finally, Issuance assumes that the Client and Mediator have a secure and
+### Mediator State {#mediator-state}
+
+Issuance assumes that Mediators maintain local state for each Client. The mechanism
+of identifying a Client is specific to each Mediator, and is not defined in this document.
+As examples, the Mediator could use device-specific certificates or account authentication
+to identify a Client.
+
+Mediators are expected to know the ISSUER_POLICY_WINDOW for any ISSUER_NAME to which
+they allow access. This information can be retrieved using the URIs defined in {{setup}}.
+
+For each pair of Client and Issuer, the Mediator maintains a specific policy window
+start and end time for each Issuer requested by the Client. Associated with this window,
+the Mediator determines a CLIENT_GROUP_ID value for the Client-Issuer pair, which
+changes every time the policy window elapses. The Mediator additionally maintains
+a counter of successful tokens issued within the policy window for each ANON_ORIGIN_ID
+requested by the Client, as well as a GROUP_ORIGIN_ID that it received for each
+ANON_ORIGIN_ID.
+
+The Mediator assigns CLIENT_GROUP_ID values to Clients in way that segments them into
+groups of roughly equal size. For example, at any time, Clients could be spread
+across 10 different CLIENT_GROUP_ID values. The specific values for the CLIENT_GROUP_ID
+are integers determined by the Mediator, and is not reused across different
+policy windows in a short period of time. One simple mechanism would be to have a counter
+for groups, up to the maximum value of uint64, before wrapping around.
+
+A specific CLIENT_GROUP_ID can be in one of three states:
+
+- Open: Clients can be assigned this CLIENT_GROUP_ID upon first access to a given
+Issuer for their specific policy window. This state lasts exactly the number
+of seconds in ISSUER_POLICY_WINDOW.
+- Draining: Clients can no longer be assigned this CLIENT_GROUP_ID for new access,
+but some Clients are still using this CLIENT_GROUP_ID. This state also lasts exactly
+the number of seconds in ISSUER_POLICY_WINDOW.
+- Closed: No Clients are using this CLIENT_GROUP_ID, and no new Clients can have
+have it assigned.
```

Comment:
When should this be reaped? Perhaps after a large enough period of time, say ten times the policy window period?